File: types/learning-v2/chapter-plan.ts
// types/learning-v2/chapter-plan.ts
// V2 章节计划：章节目标 → 3～7 个短任务的拆分契约
// 依据 docs/architecture/v2_课程生成逻辑.md §2.2

import type { GenerationMeta } from './generation-meta';

export type ChapterPlanStatus = 'draft' | 'active' | 'completed' | 'superseded';

export type TaskTeachingPattern =
  | 'explain'
  | 'worked_example'
  | 'compare'
  | 'process'
  | 'derive'
  | 'practice'
  | 'case_analysis'
  | 'recap';

export type TaskEvidencePolicy = 'none' | 'self_report' | 'checkpoint';

export type TaskOrigin = 'initial' | 'remediation' | 'extension' | 'user_requested';

export type PlannedTaskStatus = 'planned' | 'prefetched' | 'active' | 'completed' | 'skipped';

export interface PlannedTask {
  taskId: string;
  order: number;
  title: string;
  objectiveId: string;
  taskGoal: string;
  observableOutcome: string;
  conceptKeys: string[];
  prerequisiteTaskIds: string[];
  teachingPattern: TaskTeachingPattern;
  expectedMinutes: number;
  evidencePolicy: TaskEvidencePolicy;
  origin: TaskOrigin;
  status: PlannedTaskStatus;
}

export interface ChapterPlan {
  chapterId: string;
  planId: string;
  planVersion: number;
  objectiveIds: string[];
  tasks: PlannedTask[];
  status: ChapterPlanStatus;
  createdAt: number;
  updatedAt: number;
  generationMeta: GenerationMeta;
  /** P4：已应用的计划补丁数（每章上限 MAX_PLAN_PATCHES_PER_CHAPTER） */
  appliedPatchCount?: number;
}

// ---- 计划校验结果（拦截项见文档 §2.2「计划校验必须拦截」） ----

export interface PlanValidationIssue {
  code: string;
  taskId?: string;
  message: string;
}

export interface PlanValidationResult {
  valid: boolean;
  errors: PlanValidationIssue[];
  warnings: PlanValidationIssue[];
}

// ---- P4 动态调度：计划补丁 ----

/** 补丁操作白名单（§P4）：只允许未展示任务的结构化调整 */
export type PlanPatchOperationType =
  | 'insert_task'      // 插入前置/补救任务
  | 'skip_task'        // 压缩：跳过冗余任务（基于强先验证据）
  | 'reorder_tasks'    // 重排剩余任务顺序
  | 'adjust_depth';    // 调整任务深度（简要/详细）

export interface PlanPatchOperation {
  type: PlanPatchOperationType;
  /** insert_task 新任务数据（其余操作为目标 taskId） */
  newTask?: PlannedTask;
  /** 目标任务（skip/reorder/adjust_depth） */
  targetTaskIds: string[];
  /** reorder: 新顺序（仅剩余任务的 taskId 序列） */
  newOrder?: string[];
  /** adjust_depth: 目标深度 */
  depth?: 'brief' | 'standard' | 'detailed';
}

/** 计划补丁：版本化、可追溯、可拒绝旧结果（P4 §完成定义） */
export interface ChapterPlanPatch {
  patchId: string;
  /** 补丁基于的计划版本（应用时必须与当前 planVersion 一致） */
  basePlanVersion: number;
  operations: PlanPatchOperation[];
  /** 用户可见的简短调整说明（非技术化） */
  summary: string;
  /** 触发原因代码 */
  reasonCode: 'STRONG_PRIOR_EVIDENCE' | 'USER_REQUESTED_DEPTH' | 'DEPENDENCY_DISCOVERED' | 'REMEDIATION_GAP';
  confidence: number;
  createdAt: number;
}

/** 每章动态调整上限（§P4「每章动态调整次数有上限」） */
export const MAX_PLAN_PATCHES_PER_CHAPTER = 2;
/** 高影响调整（insert/skip）的置信度阈值 */
export const HIGH_IMPACT_CONFIDENCE_THRESHOLD = 0.8;

File: python-agent/prompts/chapter_plan.py
# prompts/chapter_plan.py — V2 章节任务计划生成 prompt

PROMPT = {
    "role": """<role>
你是一位学习任务规划师，负责把一个章节拆成一系列可依次完成的短学习任务。
每个任务只聚焦一个小目标，学习者花 2～6 分钟就能完成，完成后能获得明确的进展感。
</role>""",

    "chapter_context": """<chapter_context>
## 课程主题
{topic}

## 当前章节
- 标题：{chapter_title}
- 教学目标：{teaching_goal}

## 本章课程目标（任务的 objectiveId 只能从中选择）
{objectives_section}

## 学习者起点
{learner_section}
</chapter_context>""",

    "output_format": """<output_format>
## 任务拆分要求
- 把章节拆成 3～7 个短任务，按学习顺序排列
- 任务标题必须具体、指向明确动作，严禁照抄章节标题
- 每个任务只聚焦一个小目标，任务之间前后衔接自然
- 每个任务绑定的课程目标（objectiveId）必须来自上方本章课程目标列表
- 所有本章课程目标都要被任务覆盖到
- 结合学习者起点：已确认掌握的内容不必从头讲起，潜在缺口要顺带补齐
- teachingPattern 从以下枚举中单选：
  - explain（概念讲解）
  - worked_example（例题示范）
  - compare（对比辨析）
  - process（操作步骤）
  - derive（原理推导）
  - practice（练习巩固）
  - case_analysis（案例分析）
  - recap（总结回顾）
- 教学模式尽量多样化，避免全章同一模式

## 输出格式
{{
  "tasks": [
    {{
      "taskId": "t1",
      "order": 1,
      "title": "任务标题（建议 12 字内，不与章节标题重复）",
      "objectiveId": "本章课程目标 ID",
      "taskGoal": "这个任务要完成的一个小目标",
      "observableOutcome": "学习者完成后能做到的可观察结果",
      "conceptKeys": ["涉及的核心概念"],
      "prerequisiteTaskIds": [],
      "teachingPattern": "explain",
      "expectedMinutes": 3,
      "evidencePolicy": "none",
      "origin": "initial",
      "status": "planned"
    }}
  ]
}}

## 字段约定
- taskId 按顺序取 t1、t2……，与 order 保持一致
- prerequisiteTaskIds 填线性依赖（前一个任务的 taskId），首个任务为空数组
- expectedMinutes 为 2～6 的整数
- evidencePolicy 固定 "none"、origin 固定 "initial"、status 固定 "planned"

## JSON 书写规范
- 字符串值内部不要出现英文双引号，引述内容改用中文引号「」

只返回 JSON。
</output_format>""",
}
